snf-cyclades-app/synnefo/userdata/models_factory.py

```diff
@@ -0,0 +1,37 @@
+#
+# Copyright (C) 2010-2016 GRNET S.A.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+import factory
+from synnefo.userdata import models
+
+
+class PublicKeyPairFactory(factory.DjangoModelFactory):
+    FACTORY_FOR = models.PublicKeyPair
+    name = 'keypair-d20a3d59-9433-4b79-8726-20b431d89c78'
+    content = ('ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCkF3MX59OrlBs3dH5CU7lNm',
+               'vpbrgZxSpyGjlnE8Flkirnc/Up22lpjznoxqeoTAwTW034k7Dz6aYIrZGmQwe',
+               '2TkE084yqvlj45Dkyoj95fW/sZacm0cZNuL69EObEGHdprfGJQajrpz22NQoC',
+               'D8TFB8Wv+8om9NH9Le6s+WPe98WC77KLw8qgfQsbIey+JawPWl4O67ZdL5xry',
+               'puRjfIPWjgy/VH85IXg/Z/GONZ2nxHgSShMkwqSFECAC5L3PHB+0+/12M/iik',
+               'datFSVGjpuHvkLOs3oe7m6HlOfluSJ85BzLWBbvva93qkGmLg4ZAc8rPh2O+Y',
+               'IsBUHNLLMM/oQp')
+    fingerprint = '7e:eb:ab:24:ba:d1:e1:88:ae:9a:fb:66:53:df:d3:bd'
+    from datetime import datetime
```

Comment:
Move `from datetime import datetime` out of the class, otherwise you get:
```
Traceback (most recent call last):
  File "/var/tmp/synnefo/snf-cyclades-app/synnefo/api/tests/keypairs.py", line 57, in setUp
    name=self.u1_key_name)
  File "/usr/local/lib/python2.7/dist-packages/factory/base.py", line 82, in __call__
    return cls.create(**kwargs)
  File "/usr/local/lib/python2.7/dist-packages/factory/base.py", line 585, in create
    return cls._generate(True, attrs)
  File "/usr/local/lib/python2.7/dist-packages/factory/base.py", line 510, in _generate
    obj = cls._prepare(create, **attrs)
  File "/usr/local/lib/python2.7/dist-packages/factory/base.py", line 485, in _prepare
    return cls._create(model_class, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/factory/django.py", line 153, in _create
    return manager.create(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/django/db/models/manager.py", line 92, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/django/db/models/query.py", line 370, in create
    obj = self.model(**kwargs)
  File "/usr/lib/python2.7/dist-packages/django/db/models/base.py", line 452, in __init__
    raise TypeError("'%s' is an invalid keyword argument for this function" % list(kwargs)[0])
TypeError: 'datetime' is an invalid keyword argument for this function
```